compiler/vm/vmrunner.nim

```diff
@@ -379,16 +380,18 @@ proc main*(args: seq[string]): int =
 
   # the execution part. Set up a thread and run it until it either exits
   # normally or abnormally
-  var thread = initVmThread(c, entryPoint.start, frame)
-  block:
+  var
+    thread = initVmThread(c, entryPoint.start, frame)
+    continueExecution = true ## whether we continue to execute after yield
+  while continueExecution:
+    continueExecution = false # default to stop execution on any yield
```

Comment:
I reworked it where it by default `break`s at the end of an iteration, then use a continue to override it within a branch. Seems a little bit better. All this to avoid infinite loop errors, I might be being silly.